aiidalab_widgets_base/viewers.py

```diff
@@ -670,38 +664,50 @@ def highlight_atoms(
     def _default_supercell(self):
         return [1, 1, 1]
 
-    @default("selection")
-    def _default_selection(self):
-        return []
+    @observe("input_selection")
+    def _observe_input_selection(self, value):
+        if value["new"] is None:
+            return
+
+        # Exclude everything that is beyond the total number of atoms.
+        selection_list = [x for x in value["new"] if x < self.natom]
 
-    @validate("selection")
-    def _validate_selection(self, provided):
-        return list(provided["value"])
+        # In the case of a super cell, we need to multiply the selection as well
+        multiplier = sum(self.supercell) - 2
+        selection_list = [
+            x + self.natom * i for x in selection_list for i in range(multiplier)
+        ]
 
-    @observe("selection")
-    def _observe_selection(self, _=None):
-        self.highlight_atoms(self.selection)
-        self._selected_atoms.value = list_to_string_range(self.selection, shift=1)
+        self.displayed_selection = selection_list
 
-        # if atom is selected from nglview, shift to selection tab
```

Comment:
@yakutovicha why was this code deleted?

Reply:
I don't remember the exact motivation. I think it is because the trait doesn't have to be set by nglview. If it is set elsewhere, the switching will still occur, which might be unwanted.

Reply:
Ok, that makes sense. Thanks!